libPhoneNumber-GeocodingParser/libPhoneNumber-GeocodingParser/SQLiteDatabaseConnection.m

```diff
@@ -0,0 +1,121 @@
+//
+//  SQLiteDatabaseConnection.m
+//  libPhoneNumber-GeocodingParser
+//
+//  Created by Rastaar Haghi on 7/1/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import "SQLiteDatabaseConnection.h"
+
+@implementation SQLiteDatabaseConnection {
+@private
+  NSString *_databasePath;
+  sqlite3 *_DB;
+  sqlite3_stmt *_insertStatement;
+}
+
+static NSString *const insertPreparedStatement = @"INSERT INTO geocodingPairs%@"
+                                                 @"( "
+                                                 @"NATIONALNUMBER, "
+                                                 @"DESCRIPTION"
+                                                 @")"
+                                                 @"VALUES "
+                                                 @"("
+                                                 @"?,"
+                                                 @"?"
+                                                 @")";
+
+static NSString *const createTablePreparedStatement =
+    @"CREATE TABLE IF NOT EXISTS geocodingPairs%@ (ID INTEGER PRIMARY KEY "
+    @"AUTOINCREMENT, "
+    @"NATIONALNUMBER TEXT, DESCRIPTION TEXT)";
+static NSString *const createIndexStatement =
+    @"CREATE INDEX IF NOT EXISTS nationalNumberIndex ON "
+    @"geocodingPairs%@(NATIONALNUMBER)";
+
+- (instancetype)initWithCountryCode:(NSString *)countryCode
+                       withLanguage:(NSString *)language {
+  self = [super init];
+  if (self != nil) {
+    NSArray *directoryPath = NSSearchPathForDirectoriesInDomains(
+        NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *documentDirectory = directoryPath[0];
+    NSString *databasePath = [[NSString alloc]
+        initWithString:[NSString stringWithFormat:@"%@/geocoding/%@.db",
+                                                  documentDirectory, language]];
+
+    sqlite3_open([databasePath UTF8String], &_DB);
+
+    [self createTable:countryCode];
+    sqlite3_prepare_v2(_DB,
+                       [[NSString stringWithFormat:insertPreparedStatement,
+                                                   countryCode] UTF8String],
```

Comment:
nit: Extract this into a variable. Clearer to read.

Reply:
fixed for clarity